packages/core/ios/Sources/JsRefreshRate.m

```diff
@@ -0,0 +1,104 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+// This ensures we can use [_bridge dispatchBlock]
+#import <React/RCTBridge+Private.h>
+#import "JsRefreshRate.h"
+
+typedef void (^frame_time_callback)(double frameTime);
+
+@implementation JsRefreshRate {
+  CADisplayLink *_jsDisplayLink;
+  NSTimeInterval _lastFrameTimestamp;
+  frame_time_callback _frameTimeCallback;
+  BOOL _isStarted;
+}
+
+@synthesize bridge = _bridge;
+// This ensures we can use [_bridge dispatchBlock]
+RCT_EXPORT_MODULE()
+
+// The plugin is initialized twice: the first time by React (as we registered this with RCT_EXPORT_MODULE),
+// and that is the moment the bridge is set.
+// The second time in DdSdk when we start it. This singleton pattern ensures we only have one instance running and
+// it is initialized.
+static JsRefreshRate *_pluginSingleton = nil;
+- (instancetype)init {
+  self->_isStarted = false;
+
+  if (!_pluginSingleton) {
+    self = [self initSingleton];
+    _pluginSingleton = self;
+  }
+
+  return _pluginSingleton;
+}
+
+- (instancetype)initSingleton {
+  self = [super init];
+
+  self->_lastFrameTimestamp = -1;
+
+  return self;
+}
```

Comment:
In objc, the `init`  and all `init*` prefixed methods must return a new instance of the class. Here an idiomatic way of making a singleton in objc:

```objective-c
static JsRefreshRate *_shared = nil;

+ (JsRefreshRate *)shared {
  if (!_shared) {
    _shared = [[JsRefreshRate alloc] init];
  }

  return _shared;
}

- (instancetype)init {
  self = [super init];
  if (self) {
    self->_lastFrameTimestamp = -1;
  }
  return self;
}
```